test/spec/modules/weboramaRtdProvider_spec.js

```diff
@@ -153,6 +153,289 @@ describe('weboramaRtdProvider', function() {
         });
       });
 
+      it('should use asset id when available and set gam targeting and send to bidders by default', function() {
+        let onDataResponse = {};
+        const moduleConfig = {
+          params: {
+            weboCtxConf: {
+              token: 'foo',
+              assetID: 'datasource:docId',
+              targetURL: 'https://prebid.org',
+              onData: (data, meta) => {
+                onDataResponse = {
+                  data: data,
+                  meta: meta,
+                };
+              },
+            }
+          }
+        };
+        const data = {
+          webo_vctx: ['foo', 'bar'],
+        };
+        const adUnitCode = 'adunit1';
+        const reqBidsConfigObj = {
+          ortb2Fragments: {
+            global: {},
+            bidder: {}
+          },
+          adUnits: [{
+            code: adUnitCode,
+            bids: [{
+              bidder: 'smartadserver'
+            }, {
+              bidder: 'pubmatic'
+            }, {
+              bidder: 'appnexus'
+            }, {
+              bidder: 'rubicon'
+            }, {
+              bidder: 'other'
+            }]
+          }]
+        };
+
+        const onDoneSpy = sinon.spy();
+
+        expect(weboramaSubmodule.init(moduleConfig)).to.be.true;
+        weboramaSubmodule.getBidRequestData(reqBidsConfigObj, onDoneSpy, moduleConfig);
+
+        let request = server.requests[0];
```

Comment:
@patmmccann I can't see the issue.
what do you suggest? 

I can see the usage of Promises on test/spec/modules/oneKeyRtdProvider_spec.js in same context. Perhaps this is a better approach?

Reply:
let's await @dgirardi response; I am not expert on if this is dangerous or not. I have seen some similar behavior causing problems but I am not sure if this behavior will cause a problem

Reply:
Because all tests share one single global mock for XHR, a test that causes deferred network activity after it has completed is likely to break other unrelated tests that do not expect anyone else to be messing with the XHR mock. This PR looks good to me, I don't see any `setTimeout` or other deferral that would cause the issue.

Reply:
hey @dgirardi your avatar is the Eternauta ?

Reply:
even for Latin America

Reply:
@patmmccann can we resolve this?

Reply:
apologies for the delay, I was traveling. Merged!